userspace/libsinsp/sinsp.h

```diff
@@ -251,7 +245,7 @@ class SINSP_PUBLIC sinsp
 	  @throws a sinsp_exception containing the error string is thrown in case
 	   of failure.
 	*/
-	void open(uint32_t timeout_ms = SCAP_TIMEOUT_MS);
+	virtual void open(uint32_t timeout_ms = SCAP_TIMEOUT_MS);
```

Comment:
I think what you have is fine for this change, but generally concretions should override concretions.  I think we should enhance this in some future change.  Rather than having:

    class sinsp
    class sinsp_mock : sinsp

We should instead have:

    class sinsp_base
    class sinsp : public sinsp_base
    class sinsp_mock : public sinsp_base

Where the methods you want to override in the mock are pure abstract `sinsp_base`

Reply:
I agree and that is the goal but won't happen in this change.